frontend/__tests__/hooks/usePlaygroundActions.test.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { renderHook, act } from "@testing-library/react";
+import { usePlaygroundActions } from "../../src/hooks/usePlaygroundActions";
+import { PlaygroundProvider } from "../../src/context/PlaygroundContext";
+
+// Mock uuid to return predictable values
+jest.mock("uuid", () => ({
+  v4: jest.fn(() => "test-uuid"),
+}));
+
+describe("usePlaygroundActions", () => {
+  // Wrap the hook in the PlaygroundProvider for testing
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <PlaygroundProvider>{children}</PlaygroundProvider>
+  );
+
+  beforeEach(() => {
+    // Clear mocks before each test
+    jest.clearAllMocks();
+  });
+
+  describe("addInputDocument", () => {
+    it("should create a new input document with the correct structure", () => {
+      const { result } = renderHook(() => usePlaygroundActions(), { wrapper });
+
+      act(() => {
+        result.current.addInputDocument("Test Document", "Test content");
+      });
+
+      // The hook should return the created document
+      expect(result.current.addInputDocument).toBeInstanceOf(Function);
+
+      // Call the function and check the returned document
+      let createdDoc;
+      act(() => {
+        createdDoc = result.current.addInputDocument(
+          "Test Document",
+          "Test content"
+        );
+      });
+
+      expect(createdDoc).toEqual({
```

Comment:
Do we need to redo the `act(...)`, seems like this could be on line 42?

Reply:
Agreed, I simplified this one.